napari/_qt/dialogs/_tests/test_qt_plugin_dialog.py

```diff
@@ -149,3 +269,91 @@ def test_constructor_visible_widgets(plugin_dialog_constructor):
+
+
+def test_version_dropdown(plugin_dialog):
+    """
+    Test that when the source drop down is changed, it displays the other versions properly.
+    """
+    widget = plugin_dialog.available_list.item(1).widget
+    assert widget.version_choice_dropdown.currentText() == "3"
+    # switch from PyPI source to conda one.
+    widget.source_choice_dropdown.setCurrentIndex(1)
+    assert widget.version_choice_dropdown.currentText() == "4.5"
+
+
+def test_plugin_list_count_items(plugin_dialog):
+    assert plugin_dialog.installed_list._count_visible() == 2
+
+
+def test_plugin_list_handle_action(plugin_dialog, qtbot):
+    item = plugin_dialog.installed_list.item(0)
+
+    plugin_dialog.installed_list.handle_action(
+        item,
+        'test-name-1',
+        InstallerActions.UPGRADE,
+    )
+
+    with patch.object(qt_plugin_dialog.WarnPopup, "exec_") as mock:
+        plugin_dialog.installed_list.handle_action(
+            item,
+            'test-name-1',
+            InstallerActions.UNINSTALL,
+        )
+        assert mock.called
+
+    item = plugin_dialog.available_list.item(0)
+    plugin_dialog.available_list.handle_action(
+        item,
+        'test-name-1',
+        InstallerActions.INSTALL,
+        version='3',
+    )
+
+    plugin_dialog.available_list.handle_action(
+        item, 'test-name-1', InstallerActions.CANCEL, version='3'
+    )
+
+    # Wait for refresh timer, state and worker to be done
+    qtbot.waitUntil(
+        lambda: not plugin_dialog._add_items_timer.isActive()
+        and plugin_dialog.refresh_state == qt_plugin_dialog.RefreshState.DONE
+    )
+    qtbot.waitUntil(lambda: not plugin_dialog.worker.is_running)
+
+
+def test_on_enabled_checkbox(plugin_dialog, qtbot, plugins, old_plugins):
+    # checks npe2 lines
+    item = plugin_dialog.installed_list.item(0)
+    widget = plugin_dialog.installed_list.itemWidget(item)
+
+    assert plugins.plugins['my-plugin'] is True
+    with qtbot.waitSignal(widget.enabled_checkbox.stateChanged, timeout=500):
+        widget.enabled_checkbox.setChecked(False)
+    assert plugins.plugins['my-plugin'] is False
+
+    # checks npe1 lines
+    item = plugin_dialog.installed_list.item(1)
+    widget = plugin_dialog.installed_list.itemWidget(item)
+
+    assert old_plugins.enabled[0] is True
+    with qtbot.waitSignal(widget.enabled_checkbox.stateChanged, timeout=500):
+        widget.enabled_checkbox.setChecked(False)
+    assert old_plugins.enabled[0] is False
+
+
+def test_add_items_outdate(plugin_dialog):
+    new_plugin = (
+        npe2.PackageMetadata(name="my-plugin", version="0.3.0"),
+        True,
+        {
+            "home_page": 'www.mywebsite.com',
+            "pypi_versions": ['0.3.0'],
+            "conda_versions": ['0.3.0'],
+        },
+    )
+    plugin_dialog._plugin_data = [new_plugin]
+    plugin_dialog._add_items()
+
+    assert 'my-plugin' not in plugin_dialog.available_set
```

Comment:
Could you update the name or add a docstring with an explanation? 
Does `my-plugin` is recognized as installed in this test?

If yes then why check the existence in `avaliable_set` and not check the presence of the update button?

Reply:
I realized now that this doesn't make any sense. I updated the test to show that the update button is visible.